src/sentry/rules/conditions/high_priority_issue.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+from typing import Optional, Sequence
+
+from sentry import features
+from sentry.eventstore.models import GroupEvent
+from sentry.models.activity import Activity
+from sentry.models.group import Group
+from sentry.rules import EventState
+from sentry.rules.conditions.base import EventCondition
+from sentry.types.activity import ActivityType
+from sentry.types.condition_activity import ConditionActivity, ConditionActivityType
+
+
+class HighPriorityIssueCondition(EventCondition):
+    id = "sentry.rules.conditions.high_priority_issue.HighPriorityIssueCondition"
+    label = "Sentry marks an issue as high priority"
+
+    def is_high_severity(self, state: EventState, group: Optional[Group]) -> bool:
+        if not group:
+            return False
+
+        try:
+            severity = float(group.get_event_metadata().get("severity", ""))
+        except (KeyError, TypeError, ValueError):
+            return False
+
+        return severity >= 0.1
+
+    def passes(self, event: GroupEvent, state: EventState) -> bool:
+        has_issue_priority_alerts = features.has("projects:high-priority-alerts", self.project)
+        if not has_issue_priority_alerts:
+            return False
+
+        is_high_severity = self.is_high_severity(state, event.group)
+        is_escalating = state.has_reappeared or state.has_escalated
```

Comment:
Are we intentionally including regressed issues here? I skimmed the spec but didn't see any mention of that specifically.

Reply:
We use `has_reappeared` for "The issue changes state from archived to escalating" in the ReappearedEventCondition.